cmd/nerdctl/builder_build_test.go

```diff
@@ -456,6 +456,51 @@ CMD ["echo", "nerdctl-build-notag-string"]
 	base.Cmd("image", "prune", "--force", "--all").AssertOK()
 }
 
+func TestBuildContextDockerImageAlias(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").AssertOK()
+	base.Cmd("image", "prune", "--force", "--all").AssertOK()
+
+	dockerfile := `FROM myorg/myapp
+CMD ["echo", "nerdctl-build-myorg/myapp"]`
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+
+	base.Cmd("build", buildCtx, fmt.Sprintf("--build-context=myorg/myapp=docker-image://%s", testutil.CommonImage)).AssertOK()
+	base.Cmd("images").AssertOutContains("<none>")
+	base.Cmd("image", "prune", "--force", "--all").AssertOK()
+}
+
+func TestBuildContextWithCopyFromDir(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").AssertOK()
+	base.Cmd("image", "prune", "--force", "--all").AssertOK()
+
+	content := "hello_from_dir_2"
+	filename := "hello.txt"
+
+	dir2 := t.TempDir()
+	filePath := filepath.Join(dir2, filename)
+	err := os.WriteFile(filePath, []byte(content), 0644)
+	assert.NilError(t, err)
+	defer os.RemoveAll(dir2)
```

Comment:
t.TempDir automatically cleans up this

Reply:
I see. Just looked at the implementation. Great to know. I've replaced it, but I just noticed, the `createBuildContext` function in this file uses os.MkdirTemp and requires a cleanup after all. 

Should I replace that in all the tests in this file, or is that out of scope for this change?

Reply:
Can be a separate PR

Reply:
I've opened this PR: https://github.com/containerd/nerdctl/pull/2869